### DISCUSSION TITLE
私以为插件无法唤起时的文案欠妥

### DISCUSSION THREAD

**Discussion by ZeroAurora** [SagerNet/SagerNet#407] [General]
现在的文案如下：
> 配置 %s 需要插件 %s，但你的专有设备供应商（通常也是监视资本主义巨头和恶意软件制造商）篡改了你的安卓系统，使该插件无法使用。

但该文案**并没有指出问题的根本所在**。
我举双手赞成该文案内的内容。但私以为，一个好的报错文案应当**清晰提示问题所在**并**引导用户解决问题**，而非谜语人一般告诉用户“你的手机厂商就是不行”而不提供解决方案。
诸如 MIUI 一类的手机系统也提供了恢复应用唤醒的方法（应用设置-自启动）。尊重用户的选择，劝说而非责骂，私以为这可以让双方的心情都更好一点。

以下是我的拙笔，也许可以有所帮助。
> 配置 %s 需要插件 %s，但你的设备供应商禁用了应用间唤醒，致使 SagerNet 无法唤醒插件并使用之。你可以尝试在插件的应用设置内允许其自启动来尝试解决此问题。
另外，此类专有设备供应商通常也是监视资本主义巨头和恶意软件制造商。他们篡改了安卓系统来达成他们不论好坏的目的。因此，我们强烈建议你尝试并使用自由的安卓系统固件。

~~（别打我呜呜呜呜）~~

**Comment by xuemind** [comment]
# 如何设置SagerNet启动Brook Plugin？

## 1. 小米MIUI 12.5.6
系统设置>应用设置>应用管理>Brook Plugin>打开自启动
<img src="https://user-images.githubusercontent.com/715345/167781458-5b1fafbd-c5ba-46e4-bb52-37d215bd595e.jpg" width="250"  />

## 2. 华为HarmonyOS 2.0.0
系统设置>应用和服务>应用启动管理>Brook Plugin>设置为手动管理
<img src="https://user-images.githubusercontent.com/715345/167781467-4ae37b89-bdcc-435d-9adf-0b44870d4e74.jpg" width="250"  />

**Reply by n374** [reply]
# 3. Vivo IQOO Origin OS
i管家 > 应用管理 > 权限管理 > XXX Plugin > X项权限 > 开启关联启动
<img src="https://user-images.githubusercontent.com/4954114/197222241-e12fd2de-957f-4dcf-8dfb-aac0dc6528e0.png" width="300">

**Comment by xuemind** [comment]
确实需要更新一下

**Comment by Lurkrazy** [comment]
虽然我很尊重开源开发者的付出，但是这文案突出一个“自以为是”

**Comment by wbbo** [comment]
是的 错误提示有点误导, 我以为只能装原生系统了.
